File: week1-c/mario/more/mario.c
#include <stdio.h>
#include <cs50.h>

// User-defined functions
void printScenario(string str, int n);

// Main code
int main(void)
{
    // Ask the user to enter a height
    int height;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Prints both pyramids until the specified height
    for (int i = 1; i <= height; i++)
    {
        printScenario(" ", height - i);
        printScenario("#", i);
        printScenario(" ", 2);
        printScenario("#", i);

        printf("\n");
    }
}

// Function to print a string n times
void printScenario(string str, int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("%s", str);
    }
}

File: week6-python/mario/more/mario.py
from cs50 import get_int


def main():

    # Ask the user to enter a height
    while True:
        height = get_int("Height: ")

        if height in range(1, 9):
            break

    # Prints both pyramid until the specified height
    for i in range(1, height + 1):
        printScenario(" ", height - i)
        printScenario("#", i)
        printScenario(" ", 2)
        printScenario("#", i)

        print()


# User-defined functions
# Function to print a string n times
def printScenario(string, n):

    for i in range(n):
        print(string, end="")


# Run the main function
if __name__ == "__main__":
    main()
